### PR TITLE
Update eslint packages

### PR DESCRIPTION
**Pull request by joecowton1**
## What are you changing?

Update packages to use 
- "@typescript-eslint/eslint-plugin": "8.1.0",
- "@typescript-eslint/parser": "8.1.0",
- "eslint": "8.57.0",

Plus a few small refactors to satisfy rule changes.

### CODE REVIEW

File: libs/@guardian/identity-auth/.eslintrc.cjs [hunk 1]
```diff
@@ -13,7 +13,9 @@ module.exports = {
 			parserOptions: {
 				tsconfigRootDir: __dirname,
 			},
-			rules: {},
+			rules: {
+				'@typescript-eslint/only-throw-error': 'off',
```
Comment (joecowton1) [inline libs/@guardian/identity-auth/.eslintrc.cjs:17]: add comment: something about public api and mention to mahesh

File: libs/@guardian/libs/.eslintrc.cjs [hunk 2]
```diff
@@ -17,7 +17,9 @@ module.exports = {
 		},
 		{
 			files: ['*.ts', '*.tsx'],
-			rules: {},
+			rules: {
+				'@typescript-eslint/prefer-promise-reject-errors': 'off',
```
Comment (joecowton1) [inline libs/@guardian/libs/.eslintrc.cjs:21]: maybe just add for line?

File: libs/@guardian/libs/src/consent-management-platform/index.test.ts [hunk 3]
```diff
@@ -128,6 +128,7 @@ describe('hotfix cmp.init', () => {
 
 				window.guCmpHotFix = {};
 				jest.resetModules();
+				// eslint-disable-next-line @typescript-eslint/no-floating-promises -- TODO
```
Comment (joecowton1) [inline libs/@guardian/libs/src/consent-management-platform/index.test.ts:131]: void the import instead here

File: package.json [hunk 4]
```diff
@@ -15,14 +15,15 @@
 		"@guardian/eslint-config-typescript": "workspace:*",
 		"@guardian/prettier": "workspace:*",
 		"@manypkg/get-packages": "2.2.0",
+		"@stylistic/eslint-plugin": "2.6.2",
```
Comment (joecowton1) [inline package.json:18]: Add to package for eslint config ts

**Review by oliverabrahams — APPROVED** [review]

**Pull request merged**